src/expressions/baseExpression.ts

```diff
@@ -1339,12 +1353,13 @@ export abstract class Expression implements Instance<ExpressionValue, Expression
   public resolveWithExpressions(expressions: Lookup<Expression>, ifNotFound: IfNotFound = 'throw'): Expression {
     return this.substitute((ex: Expression, index: int, depth: int, nestDiff: int) => {
       if (ex instanceof RefExpression) {
-        var nest = ex.nest;
+        var { nest, ignoreCase, name } = ex;
         if (nestDiff === nest) {
           var foundExpression: Expression = null;
           var valueFound = false;
-          if (hasOwnProperty(expressions, ex.name)) {
-            foundExpression = expressions[ex.name];
+          var property = ignoreCase ? findPropertyCI(expressions, name) : findProperty(expressions, name);
+          if (property) {
```

Comment:
this should prob be `if (property != null)` because it could technically be `''` (empty string)